### PR TITLE
Feature/refactor repostiry

### PR DESCRIPTION
**Pull request by joaoandre-avaiga**

### CODE REVIEW

File: src/taipy/core/_repository/_fs_base.py [hunk 1]
```diff
@@ -49,7 +50,7 @@ def object_hook(self, source):
             return source
 
 
-class _FileSystemRepository(Generic[ModelType, Entity]):
+class _FileSystemRepository(_AbstractRepository[ModelType, Entity]):
```
Comment (jrobinAV) [inline src/taipy/core/_repository/_fs_base.py:53]: I would rename the file to `_file_system_repository`. Is there any reason to keep `fs_base` name?

File: src/taipy/core/pipeline/_pipeline_repository.py [hunk 2]
```diff
@@ -23,9 +23,14 @@
 from .pipeline import Pipeline
```
Comment (jrobinAV) [inline src/taipy/core/pipeline/_pipeline_repository.py:25]: Should we delete method `_storage_folder`?

File: src/taipy/core/cycle/_cycle_repository.py [hunk 3]
```diff
@@ -16,14 +16,19 @@
 from taipy.config.config import Config
 from taipy.config.scenario.frequency import Frequency
 
-from .._repository import _FileSystemRepository
+from .._repository import _RepositoryFactory
 from ..cycle._cycle_model import _CycleModel
 from ..cycle.cycle import Cycle
 
 
-class _CycleRepository(_FileSystemRepository[_CycleModel, Cycle]):
```
Comment (jrobinAV) [inline src/taipy/core/cycle/_cycle_repository.py:24]: SHould we delete method `_storage_folder`?

File: src/taipy/core/data/_data_repository.py [hunk 4]
```diff
@@ -16,22 +16,27 @@
 
 from taipy.config.config import Config
 
+from .._repository import _RepositoryFactory
+from ..common._utils import _load_fct
 from ._data_model import _DataNodeModel
 from .data_node import DataNode
 from .generic import GenericDataNode
-from .._repository import _FileSystemRepository
-from ..common._utils import _load_fct
```
Comment (jrobinAV) [inline src/taipy/core/data/_data_repository.py:25]: SHould we delete method `_storage_folder`?

File: src/taipy/core/job/_job_repository.py [hunk 5]
```diff
@@ -16,19 +16,24 @@
...
-from .._repository import _FileSystemRepository
+from .._repository import _RepositoryFactory
 from ..common._utils import _fcts_to_dict, _load_fct
 from ..exceptions.exceptions import InvalidSubscriber
 from ..task._task_repository import _TaskRepository
+from ._job_model import _JobModel
+from .job import Job
 
 
-class _JobRepository(_FileSystemRepository[_JobModel, Job]):
```
Comment (jrobinAV) [inline src/taipy/core/job/_job_repository.py:27]: SHould we delete method `_storage_folder`?

File: src/taipy/core/task/_task_repository.py [hunk 6]
```diff
@@ -13,18 +13,23 @@
...
-from ._task_model import _TaskModel
-from .task import Task
-from .._repository import _FileSystemRepository
+from .._repository import _RepositoryFactory
 from ..common._utils import _load_fct
 from ..common.alias import TaskId
 from ..data._data_manager_factory import _DataManagerFactory
 from ..exceptions.exceptions import NonExistingDataNode
+from ._task_model import _TaskModel
+from .task import Task
```
Comment (jrobinAV) [inline src/taipy/core/task/_task_repository.py:24]: SHould we delete method `_storage_folder`?

**Review by jrobinAV — DISMISSED** [review]

File: src/taipy/core/_repository/_fs_base.py (hunk 1 above)
Comment (joaoandre-avaiga) [reply src/taipy/core/_repository/_fs_base.py:53]: Agreed.

**Review by jrobinAV — APPROVED** [review]

**Pull request merged**